experimental/PropertyDDS/packages/property-inspector-table/src/utils.tsx

```diff
@@ -627,9 +627,11 @@ export const collectionChildTableRow = (collectionPropertyProxy: BaseProxifiedPr
       const { referencedPropertyParent, relativePathFromParent } =
         PropertyProxy.getParentOfReferencedProperty(collectionProperty as ReferenceArrayProperty, propertyId);
       if (referencedPropertyParent && relativePathFromParent) {
-        property = (referencedPropertyParent as ContainerProperty).get(relativePathFromParent)!;
+        // Introducing this intermediate variable improves type inference in some newer versions of TypeScript.
+        const tmp = (referencedPropertyParent as ContainerProperty).get(relativePathFromParent)!;
+        property = tmp;
```

Comment:
This is interesting, but the 'tmp' variable isn't significantly more concise or readable.

The comment explains it sufficiently, but I personally would leave the cast, unless there's some hidden advantage that I'm missing.

Reply:
If you keep it maybe rename `tmp` to `baseProperty`?

Reply:
We get much better type safety this way than using an `as` cast. If someone added another possible type the value could have on this code path, doing an as cast here could make this code compile when its wrong, where the version I have will fail to build in such cases.

I do agree with the baseProperty rename though, so I did that.